sql/hive-thriftserver/src/main/scala/org/apache/spark/sql/hive/thriftserver/SparkSQLDriver.scala

```diff
@@ -64,9 +64,15 @@ private[hive] class SparkSQLDriver(val context: SQLContext = SparkSQLEnv.sqlCont
         new VariableSubstitution().substitute(command)
       }
       context.sparkContext.setJobDescription(substitutorCommand)
-      val execution = context.sessionState.executePlan(context.sql(command).logicalPlan)
-      hiveResponse = SQLExecution.withNewExecutionId(execution) {
+      val logicalPlan = context.sql(command).logicalPlan
```

Comment:
So this is only a problem for thriftserver.

For normal spark application, people just do `sql("DROP TABLE t")` instead of `sql("DROP TABLE t").collect()`. I think we should do the same here: if `context.sql(command).qe.logicalPlan` is a `Command`, do not call `hiveResultString`.

cc @yaooqinn

Reply:
I guess this is not a problem only for thriftserver

```
spark.sql("show tables").show

```

is normal, but produces 3 execution ids

Reply:
Ah I see. What do we see if we click the two UI links? Is one of them less reasonable than the other?

Reply:
![image](https://user-images.githubusercontent.com/8326978/112818157-ce53a280-90b5-11eb-8ac1-e4aa88f55ad5.png)

      
![image](https://user-images.githubusercontent.com/8326978/112818234-e0cddc00-90b5-11eb-9a38-c5f0c514bcda.png)

   
![image](https://user-images.githubusercontent.com/8326978/112818296-f4794280-90b5-11eb-9ddd-5f12667c37e1.png)

![image](https://user-images.githubusercontent.com/8326978/112818365-078c1280-90b6-11eb-8c81-87410e49e63e.png)

Id 1 seems useless

Reply:
This seems like a different issue, as the reported bug is the SQL execution appears twice, not 3 times.

Reply:
Basically, this looks the same to me but just different from the caller-sides. If we decide to handle them separately, I suggest we just move the `SQLExecution.withNewExecutionId` into the `hiveResultString` and then skip `Command`  from wrapping